internal/infra/powertrade/client.go

```diff
@@ -0,0 +1,251 @@
+					log.Printf("Read error: %v", err)
+				}
+				return
+			}
+
+			// Send message to buffer channel
+			msgChan <- message
+		}
+	}
+
+	log.Println("Exiting ReadMessages loop")
+}
+
+/* a function to send message to server in a thread-safe way
+to solve concurrent write error to websocket connection */
+
+func (c *PowerTradeClient) WriteMessageToServer(messageType int, data []byte) error {
+	c.writeMu.Lock()
+	defer c.writeMu.Unlock()
+	return c.conn.WriteMessage(messageType, data)
+}
+
+func (c *PowerTradeClient) SendMessage(requestMessage map[string]interface{}) error {
+
+	// Encode the requestMessage to JSON
+	message, err := json.Marshal(requestMessage)
+	if err != nil {
+		return fmt.Errorf("failed to marshal heartbeat message: %v", err)
+	}
+
+	// Send the request message to server
+	err = c.WriteMessageToServer(websocket.TextMessage, message)
+	if err != nil {
+		return fmt.Errorf("failed to write heartbeat message to server: %v", err)
+	}
+
+	log.Println("Message sent to server")
+
+	return nil
+}
+
+// send a heartbeat message to server to keep it alive
+func (c *PowerTradeClient) SendHeartbeatMessage() error {
+	timestamp := time.Now().UTC().UnixNano()
+
+	// Heartbeat message
+	heartbeatMessage := map[string]interface{}{
+		"heartbeat": map[string]string{
+			"timestamp": fmt.Sprintf("%d", timestamp),
+		},
+	}
+
+	// Encode heartbeat message to JSON
+	message, err := json.Marshal(heartbeatMessage)
+	if err != nil {
+		return fmt.Errorf("failed to marshal heartbeat message: %v", err)
+	}
+
+	// Send the heartbeat message
+	err = c.WriteMessageToServer(websocket.TextMessage, message)
+	if err != nil {
+		return fmt.Errorf("failed to write heartbeat message to server: %v", err)
+	}
+
+	log.Println("Heartbeat sent to server")
+
+	return nil
+}
+
+// send heartbeat message to Server every 10 seconds
+func (c *PowerTradeClient) startHeartbeat() {
+	for {
+		select {
+		case <-c.done:
+			return
+		case <-c.ticker.C:
+			err := c.SendHeartbeatMessage()
+			if err != nil {
+				log.Printf("Error sending heartbeat: %v", err)
+			}
+		}
+	}
+}
+
+func (c *PowerTradeClient) Stop() {
+	c.mu.Lock()
+	c.isRunning = false
+	c.mu.Unlock()
+
+	close(c.done)
+	c.ticker.Stop()
+	c.conn.Close()
+
+	log.Println("Connection to PowerTrade Websocket is closed")
+}
+
+// generate a JWT token to establish a connection with powertrade websocket
+func GenerateCredentialsSecret() string {
+	apiKey := os.Getenv("POWER_TRADE_API_KEY")
+	secretKey := os.Getenv("POWER_TRADE_SECRET_KEY")
+	now := time.Now()
+
+	claims := entities.ClientCredentialsClaim{
+		Uri:    os.Getenv("POWER_TRADE_URL"),
+		Client: "api",
+		Nonce:  now.Unix(),
+		StandardClaims: jwt.StandardClaims{
+			ExpiresAt: now.UTC().Add(1 * time.Hour).Unix(),
+			IssuedAt:  now.Unix(),
+			Subject:   apiKey,
+		},
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodES256, claims)
+	privateKey, err := jwt.ParseECPrivateKeyFromPEM([]byte(secretKey))
+	if err != nil {
+		panic(err)
```

Comment:
```suggestion
		panic(err)
```
remove panic we dont want to crash our app